### PR TITLE
Miscellaneous changes for ipam docker integration

### PR DESCRIPTION
**Pull request by aboch**
- Do not allow empty ipam type setting to overwrite default
- NetworkInfo to provide IPAM Configuration information

### CODE REVIEW

File: network.go [hunk 1]
```diff
@@ -1060,3 +1097,27 @@ func (n *network) Labels() map[string]string {
...
+	v4L := make([]*IpamConf, len(n.ipamV4Config))
+	v6L := make([]*IpamConf, len(n.ipamV6Config))
+
+	for i, c := range n.ipamV4Config {
+		cc := &IpamConf{}
+		c.CopyTo(cc)
+		v4L[i] = cc
+	}
+
+	if len(n.ipamV6Config) > 0 {
```
Comment (mavenugo) [inline network.go]: I dont think there is a need for this check

File: network.go [hunk 2]
```diff
@@ -284,12 +305,26 @@ func (n *network) CopyTo(o datastore.KVObject) error {
...
+		dstN.ipamV4Config = append(dstN.ipamV4Config, dstV4Conf)
+	}
+
 	for _, v4info := range n.ipamV4Info {
 		dstV4Info := &IpamInfo{}
 		v4info.CopyTo(dstV4Info)
 		dstN.ipamV4Info = append(dstN.ipamV4Info, dstV4Info)
 	}
 
+	if n.ipamV6Config != nil {
```
Comment (mavenugo) [inline network.go]: we dont need this either

Comment (aboch) [inline network.go]: Here we do, in datastore object should be the same as in cache. A nil list of v6 ipam config has to remain nil, not an empty list.

File: network.go (hunk 1 above)
Comment (aboch) [inline network.go]: You are right, removing here

File: network.go (hunk 2 above)
Comment (aboch) [inline network.go]: I get now what you are saying. Yeah, we do not need it here and for v6 info as well, this is in the copy function.
I got confused with what we do in the marshalling/unmarshalling functions.

Thanks, I will change it for v6 conf and info

**Comment by mavenugo** [comment]
Thanks. LGTM

**Comment by mrjana** [comment]
LGTM

**Pull request merged**